app/src/main/java/com/expidev/gcmapp/support/v4/content/MeasurementDetailsLoader.java

```diff
@@ -0,0 +1,96 @@
+package com.expidev.gcmapp.support.v4.content;
+
+import android.content.Context;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.expidev.gcmapp.db.MeasurementDao;
+import com.expidev.gcmapp.model.measurement.MeasurementDetails;
+
+import org.ccci.gto.android.common.support.v4.content.AsyncTaskBroadcastReceiverLoader;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Locale;
+
+/**
+ * Created by William.Randall on 2/23/2015.
+ */
+public class MeasurementDetailsLoader extends AsyncTaskBroadcastReceiverLoader<MeasurementDetails>
+{
+    private final MeasurementDao measurementDao;
+
+    @Nullable
+    private String measurementId;
+    @Nullable
+    private String ministryId;
+    @Nullable
+    private String mcc;
+    @Nullable
+    private String period;
+
+    public MeasurementDetailsLoader(
+        @NonNull final Context context,
+        @Nullable final Bundle args,
+        @NonNull final IntentFilter... filters)
+    {
+        this(
+            context,
+            args != null ? args.getString("measurementId") : null,
+            args != null ? args.getString("ministryId") : null,
+            args != null ? args.getString("mcc") : null,
+            args != null ? args.getString("period") : null,
+            filters
+        );
+    }
+
+    public MeasurementDetailsLoader(
+        @NonNull final Context context,
+        @Nullable String measurementId,
+        @Nullable String ministryId,
+        @Nullable String mcc,
+        @Nullable String period,
+        @NonNull final IntentFilter... filters)
+    {
+        super(context, filters);
```

Comment:
you should add in an IntentFilter to listen for MeasurementDetails changing in the background. I had forgotten it in ChurchesLoader originally, but fixed it in my current PR.

it should look something like: `addIntentFilter(BroadcastUtils.updateChurchesFilter())` where the filter matches the intent being broadcast when measurements are updated.

This will allow the Loader to automatically refresh data because it is notified the data changed.